doc/GUIDE.md

```diff
@@ -0,0 +1,186 @@
+# Using libfsm for High-Performance Pattern Matching
+
+libfsm compiles regular expressions to deterministic finite state machines (FSMs) and generates executable code. FSM-based matching runs in **linear time O(n)** with **no backtracking**.
+
+**libfsm is not a drop-in replacement for traditional regex engines.** It only supports patterns that can be compiled to FSMs.
+
+## What libfsm Cannot Do
+
+These PCRE features will not compile:
+
+* Word boundaries (`\b`)
+* Non-greedy quantifiers (`*?`, `+?`, `??`)
+* Group capture and backreferences
+* Lookahead/lookbehind assertions (`(?=`, `(?!`, `(?<=`, `(?<!`)
+* Conditional expressions (`(?(condition)then|else)`)
+* Recursion and subroutines (`(?R)`, `(?1)`)
+
+---
+
+## Quick Start
+
+Generate a matcher from a regex:
+
+```bash
+# Generate a Go matcher
+re -p -r pcre -l go -k str 'user\d+' > user_detector.go
+```
+
+This produces a standalone matcher function.
+
+---
+
+## Supported Code Generation Targets
+
+libfsm provides stable, “first-class” code generation for:
+
+| Category             | Output                         |
+| -------------------- | ------------------------------ |
+| High-level languages | **C (via `-l vmc`), Go, Rust** |
+| Toolchains           | **LLVM IR**                    |
+| Virtualization       | **Native WebAssembly**         |
+
+> Adding code generation for new languages is template-driven and straightforward.
+
+---
+
+## Workflow Overview
+
+libfsm provides two main tools: **`re`** takes patterns from command line, **`rx`** takes patterns from file.
+
+### 1. Validate the Regex
+
+Test behavior using any PCRE-compatible tool (e.g., [https://regex101.com/](https://regex101.com/)).
+
+### 2. Verify libfsm Compatibility
+
+```bash
+re -r pcre -l ast 'x*?'
+# Output: /x*?/:3: Unsupported operator
+
+rx -r pcre -l ast -d declined.txt 'x*?'
+# Unsupported character in declined.txt
+```
+
+If unsupported constructs exist, libfsm reports the failing location.
```

Comment:
I liked that we were saying that `:3` was the character offset within the regex, what happened to that?